go/vt/vtgate/executor_framework_test.go

```diff
@@ -496,6 +457,8 @@ func createExecutorEnv() (executor *Executor, sbc1, sbc2, sbclookup *sandboxconn
 	_ = hc.AddTestTablet(cell, "a0-c0", 1, "TestExecutor", "a0-c0", topodatapb.TabletType_PRIMARY, true, 1, nil)
 	_ = hc.AddTestTablet(cell, "c0-e0", 1, "TestExecutor", "c0-e0", topodatapb.TabletType_PRIMARY, true, 1, nil)
 	_ = hc.AddTestTablet(cell, "e0-", 1, "TestExecutor", "e0-", topodatapb.TabletType_PRIMARY, true, 1, nil)
+	// Below is needed so that SendAnyWherePlan doesn't fail
+	_ = hc.AddTestTablet(cell, "e0-", 1, "TestXBadVSchema", "-20", topodatapb.TabletType_PRIMARY, true, 1, nil)
```

Comment:
the problem seems to be related to hostname, if you change the hostname from `e0-` to `random` the test starts working.
The bug may be a test thing when the hostname is the same.

Reply:
ah good point. we should change `e0-` to `-20` to match the shard name.

Reply:
`-20` also conflicts. I guess we should make it `0`.
I tried that, the next error is 
```
--- FAIL: TestExecutorMaxPayloadSizeExceeded (0.00s)
    executor_test.go:1992:
        	Error Trace:	executor_test.go:1992
        	Error:      	Not equal:
        	            	expected: <nil>(<nil>)
        	            	actual  : *vterrors.fundamental(Code: NOT_FOUND
        	            	tablet aa-0000000010 not found
        	            	)
        	Test:       	TestExecutorMaxPayloadSizeExceeded
        	Messages:   	err should be nil
```